azure-common/azure/common/client_factory.py

```diff
@@ -49,72 +60,125 @@ def get_client_from_cli_profile(clientclass, **kwargs):
-
-    Example of file:
-
-    .. code:: text
-
-        # sample management library properties file
-        subscription=15dbcfa8-4b93-4c9a-881c-6189d39f04d4
-        client=a2ab11af-01aa-4759-8345-7803287dbd39
-        key=password
-        tenant=43413cc1-5886-4711-9804-8cfea3d1c3ee
-        managementURI=https://management.core.windows.net/
-        baseURL=https://management.azure.com/
-        authURL=https://login.windows.net/
-        graphURL=https://graph.windows.net/
+        config_dict = {
+            "clientId": "ad735158-65ca-11e7-ba4d-ecb1d756380e",
+            "clientSecret": "b70bb224-65ca-11e7-810c-ecb1d756380e",
+            "subscriptionId": "bfc42d3a-65ca-11e7-95cf-ecb1d756380e",
+            "tenantId": "c81da1d8-65ca-11e7-b1d1-ecb1d756380e",
+            "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+            "resourceManagerEndpointUrl": "https://management.azure.com/",
+            "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+            "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+            "galleryEndpointUrl": "https://gallery.azure.com/",
+            "managementEndpointUrl": "https://management.core.windows.net/"
+        }
+        client = get_client_from_json_dict(ComputeManagementClient, config_dict)
 
     .. versionadded:: 1.1.7
 
     :param clientclass: A SDK client class
-    :param str auth_path: Path to azureauth.properties
+    :param dict config_dict: A config dict.
     :return: An instanciated client
-    :raises: KeyError if AZURE_AUTH_LOCATION is not an environment variable and no path is provided
-    :raises: FileNotFoundError if provided file path does not exists
     """
-    auth_path = auth_path or os.environ['AZURE_AUTH_LOCATION']
-
-    with io.open(auth_path, 'r', encoding='utf-8') as auth_fd:
-        content_list = [line.strip() for line in auth_fd.readlines() if line[0] != '#' and '=' in line]
-    config_dict = dict([tuple(line.split('=', 1)) for line in content_list])
-
     parameters = {
-        'subscription_id': config_dict.get('subscription'),
-        'base_url': config_dict.get('baseURL'),
+        'subscription_id': config_dict.get('subscriptionId'),
+        'base_url': config_dict.get('resourceManagerEndpointUrl'),
     }
 
     if 'credentials' not in kwargs:
-        authority_url = (config_dict['authURL'] + '/' + 
-                         config_dict['tenant'])
+        authority_url = (config_dict['activeDirectoryEndpointUrl'] + '/' + 
+                         config_dict['tenantId'])
         context = adal.AuthenticationContext(authority_url, api_version=None)
         parameters['credentials'] = AdalAuthentication(
             context.acquire_token_with_client_credentials,
-            config_dict['managementURI'],
-            config_dict['client'],
-            config_dict['key']
+            config_dict['managementEndpointUrl'],
```

Comment:
@yugangw-msft not sure if you suggest the change the ADAL call to use `resourceManagerEndpointUrl`?